### ISSUE TITLE
[javascript] Unsure how to fix the failing relative locator test

### ISSUE THREAD

**Issue by seanpoulter** [SeleniumHQ/selenium#8375]
## 🐛 Bug Report

One of the recent tests for the relative locators is failing the JavaScript build in Travis CI. Could you please confirm what the expected behaviour is and if you'd like me to attempt a fix? Perhaps the author @AutomatedTester can provide some insight.

The test is [/javascript/node/selenium-webdriver/test/lib/by_test.js#L90-L102](https://github.com/SeleniumHQ/selenium/blame/df8c0b00426762fcf52d8b50461b0a81d8d6ce2a/javascript/node/selenium-webdriver/test/lib/by_test.js#L90-L102):
```js
  describe('RelativeBy', function() {
    it('marshalls the RelativeBy object', function() {
      let relative = by.withTagName("p").above(by.By.name("foobar"));
      let expected = {
        'relative': {
          'root': "p",
          'filters': [{ "kind": "near", "args": [{"name":"foobar"}] }],
//                              ^^^^^^                   ^^^^^^^^
//                              "above"                  ?
        }
      }
      assert.deepEqual(relative.marshall(), expected);
    });
  });
```

The test failure looks like the following. This is taken from Travis CI [job 342002768](https://travis-ci.com/github/SeleniumHQ/selenium/jobs/342002768#L335).

```Failures:
1) by RelativeBy marshalls the RelativeBy object
  Message:
    AssertionError: Expected values to be loosely deep-equal:
    
    {
      relative: {
        filters: [
          {
            args: [
              name(name) {
        return By.css('*[name="' + escapeCss(name) + '"]');
      } {
                using: 'css selector',
                value: '*[name="foobar"]'
              }
            ],
            kind: 'above'
          }
        ],
        root: 'p'
      }
    }
    
    should loosely deep-equal
    
    {
      relative: {
        filters: [
          {
            args: [
              {
                name: 'foobar'
              }
            ],
            kind: 'near'
          }
        ],
        root: 'p'
      }
    }
  Stack:
    error properties: Object({ generatedMessage: true, code: 'ERR_ASSERTION', actual: Object({ relative: Object({ root: 'p', filters: [ Object({ kind: 'above', args: [ By(css selector, *[name="foobar"]) ] }) ] }) }), expected: Object({ relative: Object({ root: 'p', filters: [ Object({ kind: 'near', args: [ Object({ name: 'foobar' }) ] }) ] }) }), operator: 'deepEqual' })
        at <Jasmine>
        at UserContext.<anonymous> (javascript/node/selenium-webdriver/test/lib/by_test.js:99:14)
        at <Jasmine>
        at processImmediate (internal/timers.js:439:21)
```

It is clear the `kind` should be `"above"`, but I'm not sure about the `name` and if it should be evaluated somewhere along the way. 🤔 
## To Reproduce

Run the tests for JavaScript tests

```
bazel test //javascript/node/selenium-webdriver:tests
```

## Expected behavior

* The test should pass locally and on TravisCI.
* The JavaScript build should succeed on TravisCI.
* We can use feedback from the CI build to evaluate PRs (like my PRs to follow for #8357 😉)

## Actual behaviour

See above.

## Test script or set of commands reproducing this issue

* Checkout `master`
* Install the prerequisites
* Run the JavaScript test suite

## Environment

TravisCI.

**Comment by seanpoulter** [comment]
Hey @shs96c and @mmerrell, I'm trying to clean up the JavaScript build and have turned to your friendly/relative locator implementation in Java from last year. Do you two remember what the next steps were? Was the following error expected? Thanks!

When I run the tests from the CLI:
```shell
bazel test //java/client/test/org/openqa/selenium/support/locators:RelativeLocatorTest
```

I get the following test results. These are the same as the Travis CI logs for [#6699.5](ci.com/github/SeleniumHQ/selenium/jobs/337209636#L373-L405](https://travis-ci.com/github/SeleniumHQ/selenium/jobs/337209636#L373-L405)
```
==================== Test output for //java/client/test/org/openqa/selenium/support/locators:RelativeLocatorTest:
JUnit4 Test Runner
.Jun. 03, 2020 3:04:36 A.M. org.openqa.selenium.testing.JUnit4TestBase$TraceMethodNameRule starting
INFO: >>> Starting shouldBeAbleToCombineFilters(org.openqa.selenium.support.locators.RelativeLocatorTest)
CREATING DRIVER
Jun. 03, 2020 3:04:36 A.M. org.openqa.selenium.testing.StaticResources ensureAvailable
INFO: Not in dev mode. Ignoring attempt to copy static resources
Jun. 03, 2020 3:04:36 A.M. org.openqa.selenium.testing.drivers.TestChromeDriver getService
INFO: chromedriver will log to /private/var/tmp/_bazel_spoulter/deb953adfcf39758b1fb9cb461842cad/execroot/selenium/_tmp/ef4e7ca3edac4d2cdfde33b6a62c3977/chromedriver6071101205316073740.log
Jun. 03, 2020 3:04:36 A.M. org.openqa.selenium.testing.drivers.TestChromeDriver getService
INFO: chromedriver will use log level ALL
Starting ChromeDriver 83.0.4103.39 (ccbf011cb2d2b19b506d844400483861342c20cd-refs/branch-heads/4103@{#416}) on port 45994
Only local connections are allowed.
Please see https://chromedriver.chromium.org/security-considerations for suggestions on keeping ChromeDriver safe.
ChromeDriver was started successfully.
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Jun. 03, 2020 3:04:38 A.M. org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: W3C
CREATED TestChromeDriver: chrome on MAC (29c12e8c62c7fc12113cec40bfeb732b)
Jun. 03, 2020 3:04:38 A.M. org.openqa.selenium.remote.RemoteWebDriver log
INFO: Executing: get [29c12e8c62c7fc12113cec40bfeb732b, get {url=http://localhost:42113/common/relative_locators.html}]
Jun. 03, 2020 3:04:39 A.M. org.openqa.selenium.remote.RemoteWebDriver log
INFO: Executed: (Response: SessionID: 29c12e8c62c7fc12113cec40bfeb732b, Status: 0, Value: null)
Jun. 03, 2020 3:04:39 A.M. org.openqa.selenium.testing.JUnit4TestBase$TraceMethodNameRule finished
INFO: <<< Finished shouldBeAbleToCombineFilters(org.openqa.selenium.support.locators.RelativeLocatorTest)
E.Jun. 03, 2020 3:04:39 A.M. org.openqa.selenium.testing.JUnit4TestBase$TraceMethodNameRule starting
INFO: >>> Starting shouldBeAbleToFindElementsAboveAnother(org.openqa.selenium.support.locators.RelativeLocatorTest)
CREATING DRIVER
CREATED TestChromeDriver: chrome on MAC (29c12e8c62c7fc12113cec40bfeb732b)
Jun. 03, 2020 3:04:39 A.M. org.openqa.selenium.remote.RemoteWebDriver log
INFO: Executing: get [29c12e8c62c7fc12113cec40bfeb732b, get {url=http://localhost:42113/common/relative_locators.html}]
Jun. 03, 2020 3:04:39 A.M. org.openqa.selenium.remote.RemoteWebDriver log
INFO: Executed: (Response: SessionID: 29c12e8c62c7fc12113cec40bfeb732b, Status: 0, Value: null)
Jun. 03, 2020 3:04:39 A.M. org.openqa.selenium.remote.RemoteWebDriver log
INFO: Executing: findElement [29c12e8c62c7fc12113cec40bfeb732b, findElement {using=id, value=below}]
Jun. 03, 2020 3:04:39 A.M. org.openqa.selenium.remote.RemoteWebDriver log
INFO: Executed: (Response: SessionID: 29c12e8c62c7fc12113cec40bfeb732b, Status: 0, Value: {element-6066-11e4-a52e-4f735466cecf=a6b7b930-ae0e-4fc7-94f0-be16e5fafa87})
Jun. 03, 2020 3:04:39 A.M. org.openqa.selenium.testing.JUnit4TestBase$TraceMethodNameRule finished
INFO: <<< Finished shouldBeAbleToFindElementsAboveAnother(org.openqa.selenium.support.locators.RelativeLocatorTest)
E
Time: 3.09
There were 2 failures:
1) shouldBeAbleToCombineFilters(org.openqa.selenium.support.locators.RelativeLocatorTest)
java.lang.IllegalArgumentException: Argument is of an illegal type: org.openqa.selenium.By$ByTagName
	at org.openqa.selenium.remote.internal.WebElementToJsonConverter.apply(WebElementToJsonConverter.java:85)
	at org.openqa.selenium.remote.internal.WebElementToJsonConverter.apply(WebElementToJsonConverter.java:79)
	at org.openqa.selenium.remote.internal.WebElementToJsonConverter.apply(WebElementToJsonConverter.java:79)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.openqa.selenium.remote.RemoteWebDriver.executeScript(RemoteWebDriver.java:431)
	at org.openqa.selenium.support.locators.RelativeLocator$RelativeBy.findElements(RelativeLocator.java:176)
	at org.openqa.selenium.remote.RemoteWebDriver.findElements(RemoteWebDriver.java:321)
	at org.openqa.selenium.support.locators.RelativeLocatorTest.shouldBeAbleToCombineFilters(RelativeLocatorTest.java:51)
2) shouldBeAbleToFindElementsAboveAnother(org.openqa.selenium.support.locators.RelativeLocatorTest)
java.lang.IllegalArgumentException: Argument is of an illegal type: org.openqa.selenium.By$ByTagName
	at org.openqa.selenium.remote.internal.WebElementToJsonConverter.apply(WebElementToJsonConverter.java:85)
	at org.openqa.selenium.remote.internal.WebElementToJsonConverter.apply(WebElementToJsonConverter.java:79)
	at org.openqa.selenium.remote.internal.WebElementToJsonConverter.apply(WebElementToJsonConverter.java:79)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.openqa.selenium.remote.RemoteWebDriver.executeScript(RemoteWebDriver.java:431)
	at org.openqa.selenium.support.locators.RelativeLocator$RelativeBy.findElements(RelativeLocator.java:176)
	at org.openqa.selenium.remote.RemoteWebDriver.findElements(RemoteWebDriver.java:321)
	at org.openqa.selenium.support.locators.RelativeLocatorTest.shouldBeAbleToFindElementsAboveAnother(RelativeLocatorTest.java:41)

FAILURES!!!
Tests run: 2,  Failures: 2
```

**Comment by barancev** [comment]
Duplicate of #8368

**Comment by barancev** [comment]
Sorry, it's not a duplicate, a different binding.

**Comment by AutomatedTester** [comment]
fixed in 319e105e1d15aa8974783b22aa83f2f1d39e7251
